test/e2e/policies.test-suite/topology-aware/n4c16/test06-fuzz/generate.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+
+	m "github.com/containers/nri-plugins/test/gofmbt"
+)
+
+type TestState struct {
+	cpu       int
+	mem       int
+	rescpu    int
+	podCpuMem map[string][2]int
+}
+
+func (s *TestState) String() string {
+	return fmt.Sprintf("[cpu:%d mem:%d pods:%v]", s.cpu, s.mem, s.podCpuMem)
+}
+
+func createPod(pod string, contcount, cpu, mem int) m.Body {
+	return func(current m.State) m.State {
+		s := current.(*TestState)
+		if s.cpu < cpu*contcount || s.mem < mem*contcount {
+			// refuse from state change if not enough resources
+			return nil
+		}
+		if _, ok := s.podCpuMem[pod]; ok {
+			// refuse to create pod if it is already running
+			return nil
+		}
+		newPodCpuMem := make(map[string][2]int)
+		for k, v := range s.podCpuMem {
+			newPodCpuMem[k] = v
+		}
+		newPodCpuMem[pod] = [2]int{cpu * contcount, mem * contcount}
+		return &TestState{
+			cpu:       s.cpu - cpu*contcount,
+			mem:       s.mem - mem*contcount,
+			podCpuMem: newPodCpuMem,
+		}
+	}
+}
+
+func deletePod(pod string) m.Body {
+	return func(current m.State) m.State {
+		s := current.(*TestState)
+		cpumem, ok := s.podCpuMem[pod]
+		if !ok {
+			// refuse to delete pod if it is not running
+			return nil
+		}
+		newPodCpuMem := make(map[string][2]int)
+		for k, v := range s.podCpuMem {
+			if k != pod {
+				newPodCpuMem[k] = v
+			}
+		}
+		return &TestState{
+			cpu:       s.cpu + cpumem[0],
+			mem:       s.mem + cpumem[1],
+			podCpuMem: newPodCpuMem,
+		}
+	}
+}
+
+func main() {
+	maxMem := 7500         // 7.5GB
+	maxCpu := 15000        // 15 cores
+	maxReservedCpu := 1000 // 1 core
+	maxTestSteps := 3000
```

Comment:
Should we also have something like

```suggestion
var (
	maxMem         int
	maxCpu         int
	maxReservedCpu int
	maxTestSteps   int
)

func main() {
	flag.IntVar(&maxMem, "mem", 7500, "memory available for test pods")
	flag.IntVar(&maxCpu, "cpu", 15000, "non-reserved milli-CPU available for test pods")
	flag.IntVar(&maxReservedCpu, "reserved-cpu", 1000, "reserved milli-CPU availble for test pods")
	flag.IntVar(&maxTestSteps, "test-steps", 3000, "number of test steps")
	flag.Parse()
```

and update `generate.sh` to use the pure go generator instead of the fMBT docker image ?

Reply:
Done.